pkg/proxy/proxy.go

```diff
@@ -282,12 +284,60 @@ func (r *Redirect) getDestinationInfo(dstIPPort string) EndpointInfo {
 	return info
 }
 
+const identityKey int = 0
+
+func newIdentityContext(ctx context.Context, id int) context.Context {
+	return context.WithValue(ctx, identityKey, id)
+}
+
+func identityFromContext(ctx context.Context) (int, bool) {
+	val, ok := ctx.Value(identityKey).(int)
+	return val, ok
+}
+
 // CreateOrUpdateRedirect creates or updates a L4 redirect with corresponding
 // proxy configuration. This will allocate a proxy port as required and launch
 // a proxy instance. If the redirect is aleady in place, only the rules will be
 // updated.
 func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, source ProxySource) (*Redirect, error) {
-	fwd, err := forward.New()
+	customDialer := func(ctx context.Context, network, address string) (net.Conn, error) {
+		d := &net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}
+
+		c, err := d.DialContext(ctx, network, address)
+		if err != nil {
+			return nil, err
+		}
+
+		if id, ok := identityFromContext(ctx); ok {
```

Comment:
This stores the identity in the socket's sk_mark so the BPF program can retrieve it again